### PR TITLE
ENH: Let `annex init` output pass through to give user feedback

### LINKED ISSUE #2844: `install` should relay git annex output
Getting a dataset on a slow USB stick in annex v6 mode:

```
% git clone https://github.com/psychoinformatics-de/studyforrest-data-structural.git
Cloning into 'studyforrest-data-structural'...
remote: Counting objects: 1520, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 1520 (delta 1), reused 0 (delta 0), pack-reused 1517
Receiving objects: 100% (1520/1520), 132.19 KiB | 319.00 KiB/s, done.
Resolving deltas: 100% (9/9), done.
Checking out files: 100% (213/213), done.
(datalad3-dev) 1 mih@meiner /media/mih/C799-BE4A/test % cd studyforrest-data-structural
(datalad3-dev) mih@meiner ...E4A/test/studyforrest-data-structural (git)-[master] % git annex init --version 6
init  
  Detected a filesystem without fifo support.

  Disabling ssh connection caching.

  Detected a crippled filesystem.
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)

  Entering an adjusted branch where files are unlocked as this filesystem does not support locked files.

Checking out files: 100% (209/209), done.
Switched to branch 'adjusted/master(unlocked)'
download failed: Not Found

  Remote origin not usable by git-annex; setting annex-ignore
(Auto enabling special remote mddatasrc...)
ok
(recording state in git...)
git annex init --version 6  3.65s user 4.22s system 5% cpu 2:25.90 total
```

Cloning takes few seconds, almost all the time is spent on `annex init`.

Now an attempt to do the same with datalad:

```
% DATALAD_REPO_VERSION=6 datalad install -s https://github.com/psychoinformatics-de/studyforrest-data-structural.git studyforrest-data-structural_alt      
[INFO   ] Cloning https://github.com/psychoinformatics-de/studyforrest-data-structural.git [1 other candidates] into '/media/mih/C799-BE4A/test/studyforrest-data-structural_alt' 
install(ok): /media/mih/C799-BE4A/test/studyforrest-data-structural_alt (dataset)
DATALAD_REPO_VERSION=6 datalad install -s  studyforrest-data-structural_alt  4.46s user 4.87s system 5% cpu 2:36.36 total
```

No output whatsoever for 2 min 30s. Most people would probably think the beast has died (stick, datalad, or both).

### PR DESCRIPTION
**Pull request by mih**
Fixes #2844

### CODE REVIEW

**Comment by mih** [comment]
From travis: `The job exceeded the maximum log length, and has been terminated.`

Dunno what to do about this, but I prefer user feedback on common functionality over meeting travis specs for this test run.

**Comment by bpoldrack** [comment]
We might want to disable this build altogether and only manually enable it on selected tests, if there's a failure that doesn't reproduce locally. I don't see another use of it other than debugging a failure that happens on Travis only.
Is there another reason I missed, @yarikoptic ?

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -1289,7 +1289,8 @@ def _init(self, version=None, description=None):
                                 where='local',
                                 reload=False)
 
-        self._run_annex_command('init', annex_options=opts)
+        self._run_annex_command('init', annex_options=opts,
+                                log_stdout=False, log_stderr=False, log_online=True)
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: I do not think that it is a good idea to make a special case for `init` here, while we have been so careful about shielding all other commands from dumping their output to the screen as is.  If "online" logging is needed, provide a callback for it `log_stdout=callable`, which would log it into the log or do whatever you want it to do (e.g. UI could provide some kind of one line scroller or spinning wheel or whatever).

Comment (bpoldrack) [reply datalad/support/annexrepo.py]: Sounds like the better solution. Agree.

Comment (mih) [reply datalad/support/annexrepo.py]: If you agree, it likely means you understand what it means. Can you do it?

Comment (mih) [reply datalad/support/annexrepo.py]: Keep in mind that the output of annex init includes progress reports and counters. I guess that machinery can handle that too. Next application would then be to apply this to the non-existing output of GitPython's repoi init method. That progress reporting is also causing much confusing in the same context (slow network, slow FS).

**Comment by mih** [comment]
Seems things are not so straightforward https://github.com/datalad/datalad/issues/2844#issuecomment-425096140

**Comment by mih** [comment]
FTR: #2872 shows the same violation of log size limits as this PR, despite not overlapping in scope.

**Pull request closed without merging**